server/controllers/file.controller.js

```diff
@@ -119,3 +119,25 @@ export function getFileContent(req, res) {
     res.send(resolvedFile.content);
   });
 }
+
+export function updateFileName(req, res) {
```

Comment:
i think that the file name should be extracted client side, rather than server side. therefore, the file should be updated based on whatever is in `req.body`, and if you only want to update the name, you should only send the name.

this also means i think this function should be named `updateFile`.

Reply:
I think `updateFileName` is more suitable as we are only saving file name not whole file. `updateFile` may be misleading.

Reply:
yeah i hear you! i'm thinking that this api endpoint should update the whole file, rather than just the name, and then it can be used to update just the file name, or the file content, or both, depending on what is in the body of the request. does that make sense?